tunnel.go

```diff
@@ -70,6 +77,87 @@ type TunnelUpdateParams struct {
 	Secret    string `json:"tunnel_secret,omitempty"`
 }
 
+type UnvalidatedIngressRule struct {
+	Hostname      string
+	Path          string
+	Service       string
+	OriginRequest OriginRequestConfig `yaml:"originRequest"`
+}
+
+// OriginRequestConfig is a set of optional fields that users may set to
+// customize how cloudflared sends requests to origin services. It is used to set
+// up general config that apply to all rules, and also, specific per-rule
+// config.
+// Note: To specify a time.Duration in go-yaml, use e.g. "3s" or "24h".
+// This is copied directly from: https://github.com/cloudflare/cloudflared/blob/eed7d7bbc90ccaca129bc6b6c3fd18b19e6bf856/config/configuration.go#L189
+type OriginRequestConfig struct {
+	// HTTP proxy timeout for establishing a new connection
+	ConnectTimeout *time.Duration `yaml:"connectTimeout"`
```

Comment:
we aren't using YAML anywhere so all these tags can go. best stick to the publicly documented endpoint https://api.cloudflare.com/#cloudflare-tunnel-configuration-put-configuration